helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -201,6 +201,10 @@ public ZkClient getZkClient() {
     return (ZkClient) getRealmAwareZkClient();
   }
 
+  public String getZkAddr() {
```

Comment:
I think we don't want to expose the ZkAddr to avoid unnecessary dependencies. In a multi ZK scenario, the address might be a mock address or a fake one. Or we may eventually get rid of this field.

This method is the only safe option,   public RealmAwareZkClient getRealmAwareZkClient(); However, there is no way to get "address" from a RealmAwareZkClient interface.
Although this is a problem that we have noticed for a long time, it seems that we are still facing the same problem.

IMO, the right solution is to keep refactoring the WAGED tool even WAGED rebalancer so it accepts RealmAwareZkClient instead of the raw address. This means we ensure our assumption, the physical ZK address is hidden in the RealmAwareZkClient interface, is guaranteed through all of our code.

However, I'm not sure how much will this change cost. It seems to be a must.

Reply:
TFTR. Updated with newly added API in rebalancer tool.